src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_params.py

```diff
@@ -8,28 +8,72 @@
+    (register_cli_argument, CliArgumentType, register_extra_cli_argument)
 from azure.cli.core.commands.parameters import \
     (tags_type, location_type, resource_group_name_type,
      get_resource_name_completion_list, enum_choice_list, file_type)
 
-from ._validators import \
-    (application_enabled)
+from azure.cli.command_modules.batch._validators import \
+    (application_enabled, datetime_format, storage_account_id, application_package_reference_format,
+     validate_client_parameters, validate_pool_resize_parameters, metadata_item_format,
+     certificate_reference_format, validate_json_file, validate_cert_file)
 
 # pylint: disable=line-too-long
 # ARGUMENT DEFINITIONS
 
-batch_name_type = CliArgumentType(help='Name of the Batch account.', options_list=('--account_name',), completer=get_resource_name_completion_list('Microsoft.Batch/batchAccounts'), id_part=None)
-
+batch_name_type = CliArgumentType(help='Name of the Batch account.', options_list=('--account-name',), completer=get_resource_name_completion_list('Microsoft.Batch/batchAccounts'), id_part=None)
 
 # PARAMETER REGISTRATIONS
 
-register_cli_argument('batch', 'account_name', batch_name_type, options_list=('--name', '-n'))
 register_cli_argument('batch', 'resource_group_name', resource_group_name_type, completer=None, validator=None)
+register_cli_argument('batch account', 'account_name', batch_name_type, options_list=('--name', '-n'))
 register_cli_argument('batch account create', 'location', location_type)
 register_cli_argument('batch account create', 'tags', tags_type)
+register_extra_cli_argument('batch account create', 'storage_account_name', help='The storage account name to be used for auto storage account.', validator=storage_account_id)
 register_cli_argument('batch account set', 'tags', tags_type)
+register_extra_cli_argument('batch account set', 'storage_account_name', help='The storage account name to be used for auto storage account.', validator=storage_account_id)
 register_cli_argument('batch account keys renew', 'key_name', **enum_choice_list(AccountKeyType))
 register_cli_argument('batch application', 'account_name', batch_name_type, options_list=('--name', '-n'), validator=application_enabled)
 register_cli_argument('batch application package create', 'package_file', type=file_type, help='The path of the application package in zip format', completer=FilesCompleter())
 register_cli_argument('batch location quotas show', 'location_name', location_type)
+
+for command in ['list', 'show', 'create', 'set', 'delete', 'package']:
+    register_cli_argument('batch application {}'.format(command), 'account_name', batch_name_type, options_list=('--name', '-n'), validator=application_enabled)
+
+register_cli_argument('batch pool resize', 'if_modified_since', help='Specify this header to perform the operation only if the resource has been modified since the specified date/time.', type=datetime_format, arg_group='Pre-condition')
```

Comment:
@derekbekoe - yeah these strings were copied straight from our swagger spec - we're endevouring to clean up this kind of terminology both here and in the spec itself :)